pkgs/tools/security/step-kms-plugin/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, pkg-config
+, pcsclite
+, softhsm
+, writeScriptBin }:
+
+let
+  getScriptName = "get_softhsm_so_path";
+  getSoftHSMsoPath = writeScriptBin getScriptName ''
+    #!/usr/bin/env sh
+    echo ${softhsm}/lib/softhsm/libsofthsm2.so
+  '';
+
+in buildGoModule rec {
+  pname = "step-kms-plugin";
+  version = "0.6.1";
+
+  src = fetchFromGitHub {
+    owner = "smallstep";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-SvdV/eB+VWVMKPLptGWKPey4iUwkNXTyma+VBOzWwg8=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    pcsclite
+    softhsm
+  ];
+
+  ldflags = [
+    "-w"
+    "-s"
+    "-X github.com/smallstep/step-kms-plugin/cmd.Version=${version}"
+  ];
+
+  vendorHash = "sha256-Z1hMZGRLiLlrYlyV4GBL+zCPJv+i/EcZPI07RinpY2Q=";
```

Comment:
```suggestion
  vendorHash = "sha256-U9MKRW4rUdh6FTAr9lrKplBFzKM7mLll9zvRypobkl0=";
```

Reply:
I am getting a different hash than either of yours:

```
error: hash mismatch in fixed-output derivation '/nix/store/gb1lzlha5vfwxnmkr9dgjq8chsclpp56-step-kms-plugin-0.6.1-go-modules.drv':
         specified: sha256-Z1hMZGRLiLlrYlyV4GBL+zCPJv+i/EcZPI07RinpY2Q=
            got:    sha256-AD3gxAtwWT85zQL4yuIbD/C5d5pW7+tRdxGs2S2RS5E=
```

I'm not familiar with this stuff. Could it be pulling from an unstable (different file each time) CMS or something?

Reply:
No I'm just on Linux with a normal filesystem

Reply:
So we were getting different hashes because I was using `postBuild`. Switching to `preBuild` seems to have fixed it.